From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Debayan Kar <kardebayan3@gmail.com>
Date: Sun, 25 Oct 2026 16:06:29 +0000
Subject: [PATCH 10/12] ICustomCaptureResult: Use the TotalCaptureResult
 constructor of Android 12+

toTotalCaptureResult() builds a CaptureResultExtras and a
TotalCaptureResult through reflection, with the constructors of
Android 10 and 11. Since Android 12 the TotalCaptureResult constructors
also take the logical camera ID and the long arguments of the extras
are passed as Integers, so every conversion of a post-processed result
fails with "null capture result!".

Call the constructor that takes the request ID and frame number
directly instead.
---
 .../protocol/ICustomCaptureResult.smali       | 651 ++----------------
 1 file changed, 47 insertions(+), 604 deletions(-)

diff --git a/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali b/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali
index 5bbc364f..1f48ecbc 100644
--- a/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali
+++ b/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali
@@ -261,558 +261,64 @@
 .end method
 
 .method public static toTotalCaptureResult(Lcom/xiaomi/protocol/ICustomCaptureResult;IZ)Landroid/hardware/camera2/TotalCaptureResult;
-    .locals 22
-
-    const-string v0, "|"
-
-    const-class v1, Ljava/lang/String;
+    .locals 12
 
     :try_start_0
-    invoke-virtual/range {p0 .. p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getSequenceId()I
-
-    move-result v3
-
-    invoke-virtual/range {p0 .. p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getFrameNumber()J
-
-    move-result-wide v4
-
-    sget-object v6, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
-
-    new-instance v7, Ljava/lang/StringBuilder;
-
-    invoke-direct {v7}, Ljava/lang/StringBuilder;-><init>()V
-
-    const-string v8, "toTotalCaptureResult: "
-
-    invoke-virtual {v7, v8}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    move/from16 v8, p1
-
-    invoke-virtual {v7, v8}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v7, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v7, v3}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v7, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v7, v4, v5}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v7}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v0
-
-    invoke-static {v6, v0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    const-string v0, "android.hardware.camera2.impl.CaptureResultExtras"
-
-    invoke-static {v0}, Ljava/lang/Class;->forName(Ljava/lang/String;)Ljava/lang/Class;
-
-    move-result-object v0
-
-    sget v6, Landroid/os/Build$VERSION;->SDK_INT:I
-    :try_end_0
-    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_1
-
-    const/16 v7, 0x1e
-
-    const/16 v9, 0x8
-
-    const/4 v10, 0x7
-
-    const/4 v11, 0x6
-
-    const/4 v12, 0x5
-
-    const/4 v13, 0x4
-
-    const/4 v14, 0x3
-
-    const/4 v15, 0x2
-
-    const/16 v16, 0x1
-
-    const/4 v2, 0x0
-
-    if-lt v6, v7, :cond_0
-
-    const/16 v6, 0xb
-
-    :try_start_1
-    new-array v7, v6, [Ljava/lang/Class;
-
-    sget-object v18, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v2
-
-    sget-object v18, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v16
-
-    sget-object v18, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v15
-
-    sget-object v18, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v14
-
-    sget-object v18, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v13
-
-    sget-object v18, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v12
-
-    sget-object v18, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v11
-
-    aput-object v1, v7, v10
-
-    sget-object v18, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
-
-    aput-object v18, v7, v9
-
-    sget-object v18, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
-
-    const/16 v19, 0x9
-
-    aput-object v18, v7, v19
-
-    sget-object v18, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
-
-    const/16 v20, 0xa
-
-    aput-object v18, v7, v20
-
-    invoke-virtual {v0, v7}, Ljava/lang/Class;->getDeclaredConstructor([Ljava/lang/Class;)Ljava/lang/reflect/Constructor;
-
-    move-result-object v7
-
-    new-array v6, v6, [Ljava/lang/Object;
-
-    invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v18
-
-    aput-object v18, v6, v2
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v18
-
-    aput-object v18, v6, v16
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v18
-
-    aput-object v18, v6, v15
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v18
-
-    aput-object v18, v6, v14
-
-    invoke-static {v4, v5}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
-
-    move-result-object v18
-
-    aput-object v18, v6, v13
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v18
-
-    aput-object v18, v6, v12
+    invoke-virtual {p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getSequenceId()I
 
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+    move-result v5
 
-    move-result-object v18
+    invoke-virtual {p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getFrameNumber()J
 
-    aput-object v18, v6, v11
+    move-result-wide v6
 
-    const/16 v17, 0x0
-
-    aput-object v17, v6, v10
-
-    const/16 v18, -0x1
-
-    invoke-static/range {v18 .. v18}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v21
-
-    aput-object v21, v6, v9
-
-    invoke-static/range {v18 .. v18}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v21
-
-    aput-object v21, v6, v19
-
-    invoke-static/range {v18 .. v18}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v18
-
-    aput-object v18, v6, v20
-
-    invoke-virtual {v7, v6}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v0
-    :try_end_1
-    .catch Ljava/lang/Exception; {:try_start_1 .. :try_end_1} :catch_0
-
-    goto/16 :goto_0
-
-    :catch_0
-    :try_start_2
-    sget-object v6, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
-
-    const-string v7, "could not reflect constructor with more params, try anthor"
-
-    invoke-static {v6, v7}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    new-array v6, v9, [Ljava/lang/Class;
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v2
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v16
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v15
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v14
-
-    sget-object v7, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v13
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v12
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v11
-
-    aput-object v1, v6, v10
-
-    invoke-virtual {v0, v6}, Ljava/lang/Class;->getDeclaredConstructor([Ljava/lang/Class;)Ljava/lang/reflect/Constructor;
-
-    move-result-object v7
-
-    new-array v0, v9, [Ljava/lang/Object;
-
-    invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v2
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v16
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
+    sget-object v0, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
 
-    aput-object v1, v0, v15
+    new-instance v1, Ljava/lang/StringBuilder;
 
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+    invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
 
-    move-result-object v1
+    const-string v2, "toTotalCaptureResult: "
 
-    aput-object v1, v0, v14
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
-    invoke-static {v4, v5}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
+    invoke-virtual {v1, p1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
 
-    move-result-object v1
+    const-string v2, "|"
 
-    aput-object v1, v0, v13
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+    invoke-virtual {v1, v5}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
 
-    move-result-object v1
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
 
-    aput-object v1, v0, v12
+    invoke-virtual {v1, v6, v7}, Ljava/lang/StringBuilder;->append(J)Ljava/lang/StringBuilder;
 
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
 
     move-result-object v1
 
-    aput-object v1, v0, v11
+    invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
 
-    const/4 v1, 0x0
+    if-eqz p2, :cond_0
 
-    aput-object v1, v0, v10
+    invoke-virtual {p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getMainPhysicalResult()Landroid/os/Parcelable;
 
-    invoke-virtual {v7, v0}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v0
+    move-result-object v3
 
-    goto/16 :goto_0
+    if-nez v3, :cond_1
 
     :cond_0
-    sget v6, Landroid/os/Build$VERSION;->SDK_INT:I
-
-    const/16 v7, 0x1d
-
-    if-lt v6, v7, :cond_1
-
-    new-array v6, v9, [Ljava/lang/Class;
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v2
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v16
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v15
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v14
-
-    sget-object v7, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v13
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v12
-
-    sget-object v7, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v7, v6, v11
-
-    aput-object v1, v6, v10
-
-    invoke-virtual {v0, v6}, Ljava/lang/Class;->getDeclaredConstructor([Ljava/lang/Class;)Ljava/lang/reflect/Constructor;
-
-    move-result-object v7
-
-    new-array v0, v9, [Ljava/lang/Object;
-
-    invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v2
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
+    invoke-virtual {p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getResults()Landroid/os/Parcelable;
 
-    aput-object v1, v0, v16
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v15
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v14
-
-    invoke-static {v4, v5}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
-
-    move-result-object v1
-
-    aput-object v1, v0, v13
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v12
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v11
-
-    const/4 v1, 0x0
-
-    aput-object v1, v0, v10
-
-    invoke-virtual {v7, v0}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v0
-
-    goto :goto_0
+    move-result-object v3
 
     :cond_1
-    new-array v1, v10, [Ljava/lang/Class;
-
-    sget-object v6, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v6, v1, v2
-
-    sget-object v6, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v6, v1, v16
-
-    sget-object v6, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v6, v1, v15
-
-    sget-object v6, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v6, v1, v14
-
-    sget-object v6, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
-
-    aput-object v6, v1, v13
-
-    sget-object v6, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v6, v1, v12
-
-    sget-object v6, Ljava/lang/Integer;->TYPE:Ljava/lang/Class;
-
-    aput-object v6, v1, v11
-
-    invoke-virtual {v0, v1}, Ljava/lang/Class;->getDeclaredConstructor([Ljava/lang/Class;)Ljava/lang/reflect/Constructor;
-
-    move-result-object v7
-
-    new-array v0, v10, [Ljava/lang/Object;
-
-    invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v2
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v16
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v15
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v14
-
-    invoke-static {v4, v5}, Ljava/lang/Long;->valueOf(J)Ljava/lang/Long;
-
-    move-result-object v1
-
-    aput-object v1, v0, v13
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v12
-
-    invoke-static {v2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    aput-object v1, v0, v11
-
-    invoke-virtual {v7, v0}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v0
-
-    :goto_0
-    const-class v1, Landroid/hardware/camera2/TotalCaptureResult;
-
-    invoke-virtual {v1}, Ljava/lang/Class;->getDeclaredConstructors()[Ljava/lang/reflect/Constructor;
-
-    move-result-object v1
-
-    array-length v3, v1
-
-    move v4, v2
-
-    :goto_1
-    if-ge v4, v3, :cond_3
+    invoke-static {v3}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getCameraMetaDataCopy(Ljava/lang/Object;)Ljava/lang/Object;
 
-    aget-object v5, v1, v4
-
-    invoke-virtual {v5}, Ljava/lang/reflect/Constructor;->getParameters()[Ljava/lang/reflect/Parameter;
-
-    move-result-object v6
-
-    array-length v6, v6
-
-    if-le v6, v15, :cond_2
-
-    move-object v7, v5
-
-    goto :goto_2
-
-    :cond_2
-    add-int/lit8 v4, v4, 0x1
-
-    goto :goto_1
-
-    :cond_3
-    :goto_2
-    if-eqz p2, :cond_4
-
-    sget-object v1, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
-
-    const-string v3, "prefer physical meta"
-
-    invoke-static {v1, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    invoke-virtual/range {p0 .. p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getMainPhysicalResult()Landroid/os/Parcelable;
-
-    move-result-object v1
-
-    if-nez v1, :cond_5
-
-    sget-object v1, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
-
-    const-string v3, "no physical meta, use logical"
-
-    invoke-static {v1, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    :cond_4
-    invoke-virtual/range {p0 .. p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getResults()Landroid/os/Parcelable;
-
-    move-result-object v1
-
-    :cond_5
-    invoke-static {v1}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getCameraMetaDataCopy(Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v1
+    move-result-object v3
 
-    if-nez v1, :cond_6
-
-    sget-object v0, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
+    if-nez v3, :cond_2
 
     const-string v1, "null native metadata"
 
@@ -822,110 +328,47 @@
 
     invoke-static {v0, v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
 
-    const/4 v1, 0x0
-
-    return-object v1
-
-    :cond_6
-    sget v3, Landroid/os/Build$VERSION;->SDK_INT:I
-
-    const/16 v4, 0x1c
-
-    if-ge v3, v4, :cond_7
-
-    new-array v3, v12, [Ljava/lang/Object;
-
-    aput-object v1, v3, v2
-
-    invoke-virtual/range {p0 .. p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getRequest()Landroid/hardware/camera2/CaptureRequest;
-
-    move-result-object v1
-
-    aput-object v1, v3, v16
-
-    aput-object v0, v3, v15
-
-    const/4 v1, 0x0
-
-    aput-object v1, v3, v14
-
-    invoke-static/range {p1 .. p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v0
-
-    aput-object v0, v3, v13
-
-    invoke-virtual {v7, v3}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v0
-
-    check-cast v0, Landroid/hardware/camera2/TotalCaptureResult;
+    const/4 v0, 0x0
 
     return-object v0
 
-    :cond_7
-    const-string v3, "android.hardware.camera2.impl.PhysicalCaptureResultInfo"
-
-    invoke-static {v3}, Ljava/lang/Class;->forName(Ljava/lang/String;)Ljava/lang/Class;
-
-    move-result-object v3
-
-    invoke-static {v3, v2}, Ljava/lang/reflect/Array;->newInstance(Ljava/lang/Class;I)Ljava/lang/Object;
-
-    move-result-object v3
-
-    new-array v4, v11, [Ljava/lang/Object;
-
-    aput-object v1, v4, v2
-
-    invoke-virtual/range {p0 .. p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getRequest()Landroid/hardware/camera2/CaptureRequest;
-
-    move-result-object v1
-
-    aput-object v1, v4, v16
-
-    aput-object v0, v4, v15
+    :cond_2
+    check-cast v3, Landroid/hardware/camera2/impl/CameraMetadataNative;
 
-    const/4 v1, 0x0
+    invoke-virtual {p0}, Lcom/xiaomi/protocol/ICustomCaptureResult;->getRequest()Landroid/hardware/camera2/CaptureRequest;
 
-    aput-object v1, v4, v14
+    move-result-object v4
 
-    invoke-static/range {p1 .. p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+    new-instance v1, Landroid/hardware/camera2/TotalCaptureResult;
 
-    move-result-object v0
+    const/4 v2, 0x0
 
-    aput-object v0, v4, v13
+    const/4 v8, 0x0
 
-    aput-object v3, v4, v12
+    move v9, p1
 
-    invoke-virtual {v7, v4}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
+    const/4 v10, 0x0
 
-    move-result-object v0
+    new-array v10, v10, [Landroid/hardware/camera2/impl/PhysicalCaptureResultInfo;
 
-    check-cast v0, Landroid/hardware/camera2/TotalCaptureResult;
-    :try_end_2
-    .catch Ljava/lang/Exception; {:try_start_2 .. :try_end_2} :catch_1
+    invoke-direct/range {v1 .. v10}, Landroid/hardware/camera2/TotalCaptureResult;-><init>(Ljava/lang/String;Landroid/hardware/camera2/impl/CameraMetadataNative;Landroid/hardware/camera2/CaptureRequest;IJLjava/util/List;I[Landroid/hardware/camera2/impl/PhysicalCaptureResultInfo;)V
+    :try_end_0
+    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
 
-    return-object v0
+    return-object v1
 
-    :catch_1
+    :catch_0
     move-exception v0
 
-    invoke-virtual {v0}, Ljava/lang/Exception;->printStackTrace()V
-
-    sget-object v0, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
-
-    new-instance v1, Ljava/lang/RuntimeException;
-
-    invoke-direct {v1}, Ljava/lang/RuntimeException;-><init>()V
+    sget-object v1, Lcom/xiaomi/protocol/ICustomCaptureResult;->TAG:Ljava/lang/String;
 
     const-string v2, "null capture result!"
 
-    invoke-static {v0, v2, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
+    invoke-static {v1, v2, v0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
 
-    const/4 v1, 0x0
+    const/4 v0, 0x0
 
-    return-object v1
+    return-object v0
 .end method
 
 
-- 
2.43.0

